lnrpc/signrpc/signer_server.go

```diff
@@ -872,6 +883,16 @@ func (s *Server) MuSig2CombineKeys(_ context.Context,
 func (s *Server) MuSig2CreateSession(_ context.Context,
 	in *MuSig2SessionRequest) (*MuSig2SessionResponse, error) {
 
+	// Check the now mandatory version first. We made the version mandatory,
+	// so we don't get unexpected/undefined behavior for old clients that
+	// don't specify the version. Since this API is still declared to be
+	// experimental this should be the approach that leads to the least
+	// amount of unexpected behavior.
+	version, err := UnmarshalMuSig2Version(in.Version)
```

Comment:
What about the `Sign` call? The _main_ breaking change in musig2 1.0 was that the challenge hash ends up being changed, so the signing call should be versioned as well afaict. 

Or is the idea that the context itself will now implicitly specify the version?

Reply:
Because you pass in the session ID to the `Sign` call, it calls it on the session struct in memory, which is of the correct type. So you only need to specify the version when creating the session (or when combining keys outside of a session).